sklearn/preprocessing/_function_transformer.py

```diff
@@ -40,18 +41,46 @@ class FunctionTransformer(BaseEstimator, TransformerMixin):
         kwargs forwarded. If inverse_func is None, then inverse_func
         will be the identity function.
 
-    validate : bool, optional default=True
+    validate : bool or 'array-or-frame', optional default=True
         Indicate that the input X array should be checked before calling
-        func. If validate is false, there will be no input validation.
-        If it is true, then X will be converted to a 2-dimensional NumPy
-        array or sparse matrix. If this conversion is not possible or X
-        contains NaN or infinity, an exception is raised.
+        func. The possibilities are:
+
+        - If 'array-or-frame', X will be passed through if it is a pandas
+          DataFrame or converted to a 2-dimensional array or sparse matrix. In
+          this latest case, an exception will be raised if the conversion
+          failed.
+        - If True, then X will be converted to a 2-dimensional NumPy array or
+          sparse matrix. If the conversion is not possible an exception is
+          raised.
+        - If False, there is no input validation.
+
+        When X is validated, the parameters ``accept_sparse`` and
+        ``force_all_finite`` control the validation for the sparsity and
+        the finiteness of X, respectively.
+
+        .. deprecated:: 0.20
+           ``validate=True`` as default will be replaced by
+           ``validate='array-or-frame'`` in 0.22.
+
+        .. versionchanged:: 0.20
+           ``validate`` takes the option ``'array-or-frame'``.
 
     accept_sparse : boolean, optional
         Indicate that func accepts a sparse matrix as input. If validate is
         False, this has no effect. Otherwise, if accept_sparse is false,
         sparse matrix inputs will cause an exception to be raised.
 
+    force_all_finite : boolean or 'allow-nan', optional default=True
```

Comment:
I would go for `'allow-nan'` as the default? As we are going in that direction for other transformers as well?